extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/K8sOverlordModule.java

```diff
@@ -37,31 +41,42 @@
 import org.apache.druid.tasklogs.TaskLogPusher;
 import org.apache.druid.tasklogs.TaskLogs;
 
+import java.util.Set;
+
 public class K8sOverlordModule implements DruidModule
 {
+  private Set<NodeRole> nodeRoles;
+
+  @Inject
+  public void setNodeRoles(@Self Set<NodeRole> nodeRoles)
+  {
+    this.nodeRoles = nodeRoles;
+  }
 
   @Override
   public void configure(Binder binder)
   {
-    // druid.indexer.runner.type=k8s
-    JsonConfigProvider.bind(binder, IndexingServiceModuleHelper.INDEXER_RUNNER_PROPERTY_PREFIX, KubernetesTaskRunnerConfig.class);
-    JsonConfigProvider.bind(binder, "druid.indexer.queue", TaskQueueConfig.class);
-    PolyBind.createChoice(
-        binder,
-        "druid.indexer.runner.type",
-        Key.get(TaskRunnerFactory.class),
-        Key.get(KubernetesTaskRunnerFactory.class)
-    );
-    final MapBinder<String, TaskRunnerFactory> biddy = PolyBind.optionBinder(
-        binder,
-        Key.get(TaskRunnerFactory.class)
-    );
+    if (SetUtils.emptyIfNull(nodeRoles).contains(NodeRole.OVERLORD)) {
```

Comment:
It turns out that there is an easier way to do this:

```java
@LoadScope(roles = NodeRole.OVERLORD_JSON_NAME)
public class K8sOverlordModule implements DruidModule
```

Reply:
Nice!

Reply:
neat :+1:

Reply:
oof, I even reviewed the change that added that, but totally forgot about it 🙃

Reply:
Done